docs: improve description of binding type 'latest' + fix XML examples (#4177)
camunda/camunda-docs: docs/components/best-practices/modeling/choosing-the-resource-binding-type.md

Camunda 8 supports the following binding types:

> <table>

Author: FYI: I've converted the table from Markdown format to HTML to make the code easier to read and maintain.

Reviewer: 💭 Would be helpful to split this conversion from the textual changes into different commits, to more easily see the differences in the text itself ([separate structural from behavioral changes](https://github.com/camunda/camunda/wiki/Pull-Requests-and-Code-Reviews#ive-separated-structural-from-behavioral-changes)).

Author: Ah, yes, sorry about that! I've split the commit into two now (at least for the technical writer review 😉), but they'll be squashed again on merge.

  <thead>
    <tr>
      <th align="left">Type</th>
      <th align="left">Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><code>latest</code></td>
      <td>
        <p>Resolves to the <strong>latest deployed version</strong> of the target resource at the moment the process element is activated.</p>
        <ul>
          <li>
            <p>

>               This option makes it easy to change the target resource at runtime by deploying a new version of it,
>               which allows for fast iterations during development and easy hot fixes in production:

Reviewer 2: ```suggestion
              You can use this option to easily change the target resource at runtime by deploying a new version of the resource.
              This allows for fast, iterative development, and easy hot fixes in production:
```

            </p>
            <ul>

>               <li><p>Process instances that have not yet reached the process element will directly use the newly deployed version of the target resource once the element gets activated.</p></li>

Reviewer 2: ```suggestion
              <li><p>Process instances that have not yet reached the process element use the newly deployed version of the target resource once the element is activated.</p></li>
```

>               <li><p>If the process element has already been reached, <a href="../../../concepts/process-instance-modification">process instance modification</a> can be used to re-activate it and thereby re-link it to the newly deployed version of the target resource.</p></li>

Reviewer 2: ```suggestion
              <li><p>If the process element has already been reached, you can use <a href="../../../concepts/process-instance-modification">process instance modification</a> to reactivate it and relink it to the newly deployed target resource version.</p></li>
```

            </ul>
          </li>

>           <li><p>Be aware that using <code>latest</code> can lead to unexpected behavior if you deploy a new version of the target resource without ensuring backwards compatibility with every deployed process that depends on it.</p></li>
>           <li><p>Therefore, it might not be suited for production environments that require stability and a predictable behavior.</p></li>

Reviewer 2: ```suggestion
          <li><p>Be aware that using <code>latest</code> can lead to unexpected behavior if you deploy a new version of the target resource without ensuring backwards compatibility with every deployed process that depends on it.</p><p>This might not be suited for production environments that require stability and a predictable behavior.</p></li>
```
